src/lib.rs

```diff
@@ -683,21 +695,37 @@ pub mod sync {
     ///     //   Some("Hoyten")
     /// }
     /// ```
-    #[derive(Debug)]
     pub struct Lazy<T, F = fn() -> T> {
         cell: OnceCell<T>,
-        init: F,
+        init: UnsafeCell<Option<F>>,
     }
 
+    impl<T: fmt::Debug, F: fmt::Debug> fmt::Debug for Lazy<T, F> {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            f.debug_struct("Lazy")
+                .field("cell", &self.cell)
+                .field("init", &"..")
+                .finish()
+        }
+    }
+
+    // We never create a `&F` from a `&Lazy<T, F>` so it is fine
+    // to not impl `Sync` for `F`
+    // we do create a `&mut Option<F>` in `force`, but this is
+    // properly synchronized, so it only happens once
+    // so it also does not contribute to this impl
+    // We do create a `&T` from `&Lazy<T, F>`, so `T` needs `Sync`
+    unsafe impl<T: Sync, F: Send> Sync for Lazy<T, F> {}
```

Comment:
I've missed two things here I think:

* a `Send` impl is missing
* we actually need `T: Send + Sync`  for `Sync`: If we create `Lazy` in thread1, call `force` in thread2, and drop the `Lazy` in thread1, we effectively move `T` from thread1 to thread2. 

I'll fix both